salt/config/__init__.py

```diff
@@ -2560,8 +2560,6 @@ def is_profile_configured(opts, provider, profile_name):
 
     if driver not in non_image_drivers:
         required_keys.append('image')
-    elif driver == 'vmware':
-        required_keys.append('clonefrom')
```

Comment:
I agree with you that image is different thing from guest_id but we also want to be consistent across all cloud providers. Do you think ``guest`` would make more sense than ``image``? 

Check is there to make sure that ``image`` is present for all cloud providers. In case of VMware, it checks for the presence of ``clonefrom`` argument. This is even before executing any code from the driver itself when it parses and checks if basic configuration is present. 

Some of the existing function names don't make sense in context of VMware such as list_nodes and list_images but they had to be names that way to keep the naming consistent across different cloud providers and to avoid confusion for people using salt-cloud for multiple cloud providers.